File: easy/Q2/palindrome.js
// Function to reverse the {value}
const reverse = (value) =>
  checkType(value)
    ? value.toLowerCase().split("").reverse().join("")
    : value.toString().toLowerCase().split("").reverse().join("");

// Function to check type of the {value}
const checkType = (checkValue) => typeof checkValue === "string";

// Function to check if {value} is palindrome
const isPalindrome = (string) =>
  reverse(string) === string.toString().toLowerCase();

console.log(isPalindrome("Madam")); // true
console.log(isPalindrome("Radiator")); // false
console.log(isPalindrome(1234321)); // true
console.log(isPalindrome(1234)); // false

File: easy/Q2/palindrome.py
def check_palindrome(string):
    """ Function to check if the {value} is a palindrome by keeping all letters in lowercase """
    return True if string.lower() == string.lower()[::-1] else False


def check_type(value):
    """ Function to check if the {value} is a string """
    return (
        check_palindrome(value) if type(value) == str else check_palindrome(str(value))
    )


if __name__ == "__main__":
    print(check_type("Madam"))  # True
    print(check_type("Radiator"))  # False
    print(check_type(12421))  # True
    print(check_type(1234))  # False
